design/build-clusters.md

```diff
@@ -0,0 +1,165 @@
+	
+
+As part of cluster onboarding process for above both options cluster admin should register their build cluster information with Screwdriver. Below details are required to register a build cluster.
+
+	1. Cluster name - Cluster name provided by the client
+	2. isActive - Cluster status whether its active or inactive. This will be used to route / pause. Initially this will be a manual update of cluster health.
+	3. Authorization - Authorize using user credentials and queue details.
+	4. SCM Context - git url (github.com). Applicable to only explicit build clusters which are non SD maintained.
+	5. SCM Organization - git organizations. Will be used to validate if job has permission to run on build cluster which is requesting. Applicable to only explicit build clusters which are non SD maintained.
+	6. managedByScrewdriver (true / false) - if true Cluster is maintained by screwdriver team, false cluster is maintained by external team.  
+
+
+## Design
+
+![build-clusters-design-queue.png](diagrams/build-clusters-design-queue.png)
+
+
+### Responsibilities
+	1. No change in current API to Redis implemention  
+	2. Master queue worker will push build job to queue from redis, after successfully validating blockedBy and other checks.
+	3. Build cluster queue worker will consume/poll from its respective queue.
+	4. Queue authorization (acls) will authorize the build cluster queue worker request.
+
+
+### Authentication & Authorization
+
+Master queue worker to queue - Master queue worker will be authorized with admin privileges to access all queues.  
+
+Build cluster queue worker to queue - build cluster will be registered and authorized to a queue or set of queues. Queue worker from build cluster will connect to queue using authorized user credentials and consume jobs from the queue on successful authorization.   
+
+
+### Queue setup
+TBD
+
+### Yaml 
+
+```yml
+shared:
+    environment:
+    NODE_ENV: test
+    settings:
+        email:
+    addresses: [test@email.com, test2@email.com]
+    statuses: [SUCCESS, FAILURE]
+    annotations:
+        buildCluster: iOS
+jobs:
+    main:
+        requires: [~pr, ~commit]
+        sourcePaths: ["src/app/", "screwdriver.yaml"]
+        image: node:6
+        steps:
+    - init: npm install
+    - test: npm test
+    publish:
+    requires: main
+    image: node:6
+    steps:
+        - publish: npm publish
+    ...
+```
+
+
+### New table for build cluster details
+
+Table: `buildCluster`
+
+Columns:
+
+| Name | Type | Nullable | Primary Key | Unique | Description
+| --- | --- | --- | --- | --- | --- |
+| `id` | integer | no | yes | yes | |
+| `name` | text (100) | no | no | yes | |
+| `scmOrganizations` | text(500) | yes | no | no | |
+| `scmContext` | text(200) | no | yes | no | |
+| `isActive` | boolean | no | no | no | *false or true* |
+| `managedByScrewdriver` | boolean | no | no | no | *true - cluster maintained by screwdriver team, false - cluster maintained by  external team * |
```

Comment:
Markdown for `*true - cluster maintained by screwdriver team, false - cluster maintained by  external team *` is not being evaluated properly.